src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1927,6 +1927,7 @@ internal ImmutableArray<Diagnostic> GetDiagnostics(CompilationStage stage, bool
                             var syntaxTree = syntaxTrees[i];
                             AppendLoadDirectiveDiagnostics(builder, _syntaxAndDeclarations, syntaxTree);
                             builder.AddRange(syntaxTree.GetDiagnostics(cancellationToken));
+                            builder.AddRange(syntaxTree.Options.Errors);
```

Comment:
I believe they should align with how we're reporting  CompilationOptions errors below:
builder.AddRange(Options.Errors);

We should be able to report both (deduping in the process).

---
In reply to: [101836674](https://github.com/dotnet/roslyn/pull/17113#discussion_r101836674) [](ancestors = 101836674,101836220)